src/MensaWidget/MWDataGettingWrapper.class.st

```diff
@@ -51,3 +67,83 @@ MWDataGettingWrapper class >> getTodaysMeals [
 	today := Date today yyyymmdd.
 	^ MWDataGettingWrapper getMealsOnDate: today
 ]
+
+{
+	#category : #accessing,
+	#'squeak_changestamp' : 'Oscar 7/1/2024 16:32'
+}
+MWDataGettingWrapper class >> initialLatitude [
+
+	^ InitialLatitude ifNil: [52.393822]
+]
+
+{
+	#category : #accessing,
+	#'squeak_changestamp' : 'Oscar 7/1/2024 16:21'
+}
+MWDataGettingWrapper class >> initialLatitude: anObject [
+
+	InitialLatitude := anObject
+]
+
+{
+	#category : #accessing,
+	#'squeak_changestamp' : 'Oscar 7/1/2024 16:32'
+}
+MWDataGettingWrapper class >> initialLongitude [
+
+	^ InitialLongitude ifNil: [13.129558]
+]
+
+{
+	#category : #accessing,
+	#'squeak_changestamp' : 'Oscar 7/1/2024 16:21'
+}
+MWDataGettingWrapper class >> initialLongitude: anObject [
+
+	InitialLongitude := anObject
+]
+
+{
+	#category : #accessing,
+	#'squeak_changestamp' : 'Oscar 7/1/2024 16:33'
+}
+MWDataGettingWrapper class >> initialRadius [
+
+	^ InitialRadius ifNil: [5]
+]
+
+{
+	#category : #accessing,
+	#'squeak_changestamp' : 'Oscar 7/1/2024 16:21'
+}
+MWDataGettingWrapper class >> initialRadius: anObject [
+
+	InitialRadius := anObject
+]
+
+{
+	#category : #accessing,
+	#'squeak_changestamp' : 'Oscar 7/1/2024 16:52'
+}
+MWDataGettingWrapper class >> switchLocation [
+	
+	| longitude latitude radius |
+	
+	latitude := FillInTheBlankMorph request: 'Please enter your Latitude' initialAnswer: self initialLatitude.
+	longitude := FillInTheBlankMorph request: 'Please enter your Longitude' initialAnswer: self initialLongitude.
+	radius := FillInTheBlankMorph request: 'Please enter your Radius' initialAnswer: self initialRadius.
+	self canteenDict: (MWParser transformJsonToDict: (MWApiWrapper getOutletDictFrom: latitude and: longitude andRadius: radius)).
+	self initialLongitude: longitude.
+	self initialLatitude: latitude.
+	self initialRadius: radius.
+]
+
+{
+	#category : #accessing,
+	#'squeak_changestamp' : 'Oscar 7/1/2024 16:30'
+}
+MWDataGettingWrapper class >> switchToDefaultLocation [
+	
+	self canteenDict: (MWParser transformJsonToDict: (MWApiWrapper getOutletDictFrom: 52.393822 and: 13.129558 andRadius: 5))
```

Comment:
maybe do a line break before the second bracket because otherwise it gets too long